src/testing/fast_array_utils/__init__.py

```diff
@@ -37,48 +40,245 @@ def __call__(  # noqa: D102
     _DTypeLikeFloat64 = np.dtype[np.float64] | type[np.float64]
 
 
-RE_ARRAY_QUAL = re.compile(r"(?P<mod>(?:\w+\.)*\w+)\.(?P<name>[^\[]+)(?:\[(?P<inner>[\w.]+)\])?")
+__all__ = [
+    "SUPPORTED_TYPES",
+    "SUPPORTED_TYPES_DASK",
+    "SUPPORTED_TYPES_DISK",
+    "SUPPORTED_TYPES_MEM",
+    "SUPPORTED_TYPES_MEM_DENSE",
+    "SUPPORTED_TYPES_MEM_SPARSE",
+    "ArrayType",
+    "ConversionContext",
+    "ToArray",
+]
 
 
-def get_array_cls(qualname: str) -> type[Array]:  # noqa: PLR0911
-    """Get a supported array class by qualname."""
-    m = RE_ARRAY_QUAL.fullmatch(qualname)
-    assert m
-    match m["mod"], m["name"], m["inner"]:
-        case "numpy", "ndarray", None:
-            return np.ndarray
-        case "scipy.sparse", (
-            "csr_array" | "csc_array" | "csr_matrix" | "csc_matrix"
-        ) as cls_name, None:
-            import scipy.sparse
+@dataclass
+class ConversionContext:
+    """Conversion context required for h5py."""
 
-            return getattr(scipy.sparse, cls_name)  # type: ignore[no-any-return]
-        case "cupy", "ndarray", None:
-            import cupy as cp
+    hdf5_file: h5py.File
+    get_ds_name: Callable[[], str]
 
-            return cp.ndarray  # type: ignore[no-any-return]
-        case "cupyx.scipy.sparse", ("csr_matrix" | "csc_matrix") as cls_name, None:
-            import cupyx.scipy.sparse as cu_sparse
 
-            return getattr(cu_sparse, cls_name)  # type: ignore[no-any-return]
-        case "dask.array", cls_name, _:
-            if TYPE_CHECKING:
-                from dask.array.core import Array as DaskArray
-            else:
-                from dask.array import Array as DaskArray
+@dataclass(frozen=True)
```

Comment:
Why have all of this in `__init__.py` instead of `array.py` or something?

Reply:
good point